### ISSUE TITLE
Badge Request Webpagetest

### ISSUE THREAD

**Issue by planetahuevo** [badges/shields#2717]
:clipboard: **Description**

- Which service is this badge for e.g: Webpagetest
https://www.webpagetest.org/
- What sort of information should this badge show?
This is a speed webpage test, so the info could any of the info you get from the test. 
Lets start with the simplest one: Load Time
Webpagetest | 4.34s
And maybe we could get some colours depending of the number: but that is an extra. 

We could also get:
- First Byte
- the Speed Index: https://sites.google.com/a/webpagetest.org/docs/using-webpagetest/metrics/speed-index
- Number of requests
- Size of the page... etc.

:link: **Data**

Where can we get the data from?

- Is there a public API?
Yes. 
- Does the API requires an API key?
Yes. You can get one here: 
https://www.webpagetest.org/getkey.php
- Link to the API documentation.
https://sites.google.com/a/webpagetest.org/docs/advanced-features/webpagetest-restful-apis
This post is also really useful: 
https://css-tricks.com/use-webpagetest-api/

The idea is to be able to show quickly the performance of the website. Really important these days.

Thanks

**Comment by Prouser123** [comment]
What might be a bit of an issue is that the test takes a _long_ time to complete.

Their API is also only good for 200 page loads a day, and the tests often use more than one page load. Their alternative if you need more volume is to host a private instance, which doesn't seem that feasible.

**Comment by planetahuevo** [comment]
Not sure I understand. Does the badge system have some kind of cache to prevent to reload every time?

**Comment by Prouser123** [comment]
I believe that something like the last 1000 badges are cached, but with the number of requests, badges aren't likely to stay in cache for long.

**Comment by planetahuevo** [comment]
I do not think that is a problem if we use the badge for a test already created and not for creating the test.
